### PR TITLE
Handling of queries with custom exprs

### PR DESCRIPTION
**Pull request by robdaemon**
This will better handle queries with expressions like [Discount] *
[Quantity]

### CODE REVIEW

**Comment by robdaemon** [comment]
@sbelak this is based on your feedback on the other PR for this feature

**Review by sbelak — APPROVED** [review]
Elegant

File: src/metabase/api/query_description.clj [hunk 1]
```diff
@@ -12,18 +14,21 @@
   [metadata query]
   {:table (:display_name metadata)})
 
-(defn- get-aggregation-description
+(defn- get-aggregation-details
   [metadata query]
-  (let [field-name (fn [match] (map
-                                (fn [m] (:display_name (Field (mbql.u/field-clause->id-or-literal m))))
-                                (mbql.u/match match :field-id)))]
+  (let [field-name (fn [match] (or (when (mbql.preds/Field? match)
```
[the remark this answers is not in the document]
Comment (robdaemon) [reply src/metabase/api/query_description.clj:19]: oh, probably

File: src/metabase/api/query_description.clj [hunk 2]
```diff
@@ -12,18 +14,21 @@
...
+                                   (flatten (get-aggregation-details metadata match))))]
     (when-let [agg-matches (mbql.u/match query
                              [:aggregation-options _ (options :guard :display-name)]
                              {:type :aggregation :arg (:display-name options)}
 
                              [:aggregation-options ag _]
                              (recur ag)
 
+                             [(operator :guard #{:+ :- :/ :*}) & args]
+                             (interpose (name operator) (map field-name args))
```
[the remark this answers is not in the document]
Comment (robdaemon) [reply src/metabase/api/query_description.clj:30]: I was sitting here thinking, "what I really want is to intersperse the operator between every item in this list....." and sure enough there's a Clojure function for that

**Pull request merged**